store/sqlstore/channel_store_shared_channels.go

```diff
@@ -0,0 +1,290 @@
+	}
+
+	if opts.Token != "" {
+		query = query.Where(sq.Eq{"sc.Token": opts.Token})
+	}
+
+	if opts.ExcludeHome {
+		query = query.Where(sq.NotEq{"sc.Home": true})
+	}
+
+	if opts.ExcludeRemote {
+		query = query.Where(sq.Eq{"sc.Home": true})
+	}
+
+	return query
+}
+
+// UpdateSharedChannel updates the shared channel.
+func (s SqlChannelStore) UpdateSharedChannel(sc *model.SharedChannel) (*model.SharedChannel, error) {
+	if err := sc.IsValid(); err != nil {
+		return nil, err
+	}
+
+	count, err := s.GetMaster().Update(sc)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to update shared channel with id=%s", sc.ChannelId)
+	}
+
+	if count != 1 {
+		return nil, fmt.Errorf("expected number of shared channels to be updated is 1 but was %d", count)
+	}
+	return sc, nil
+}
+
+// DeleteSharedChannel deletes a single shared channel plus associated SharedChannelRemotes.
+// Returns true if shared channel found and deleted, false if not found.
+func (s SqlChannelStore) DeleteSharedChannel(channelId string) (bool, error) {
+	transaction, err := s.GetMaster().Begin()
+	if err != nil {
+		return false, errors.Wrap(err, "DeleteSharedChannel: begin_transaction")
+	}
+	defer finalizeTransaction(transaction)
+
+	squery, args, err := s.getQueryBuilder().
+		Delete("SharedChannels").
+		Where(sq.Eq{"SharedChannels.ChannelId": channelId}).
+		ToSql()
+	if err != nil {
+		return false, errors.Wrap(err, "delete_shared_channel_tosql")
+	}
+
+	result, err := transaction.Exec(squery, args...)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to delete SharedChannel")
+	}
+
+	// Also remove remotes from SharedChannelRemotes (if any).
+	squery, args, err = s.getQueryBuilder().
+		Delete("SharedChannelRemotes").
+		Where(sq.Eq{"ChannelId": channelId}).
+		ToSql()
+	if err != nil {
+		return false, errors.Wrap(err, "delete_shared_channel_remotes_tosql")
+	}
+
+	_, err = transaction.Exec(squery, args...)
+	if err != nil {
+		return false, errors.Wrap(err, "failed to delete SharedChannelRemotes")
+	}
+
+	if err = transaction.Commit(); err != nil {
+		return false, errors.Wrap(err, "commit_transaction")
+	}
+
+	count, err := result.RowsAffected()
+	if err != nil {
+		return false, errors.Wrap(err, "failed to determine rows affected")
+	}
+
+	return count > 0, nil
+}
+
+// SaveSharedChannelRemote inserts a new shared channel remote record.
+func (s SqlChannelStore) SaveSharedChannelRemote(remote *model.SharedChannelRemote) (*model.SharedChannelRemote, error) {
+	remote.PreSave()
+	if err := remote.IsValid(); err != nil {
+		return nil, err
+	}
+
+	if err := s.GetMaster().Insert(remote); err != nil {
+		return nil, errors.Wrapf(err, "save_shared_channel_remote: channel_id=%s, id=%s", remote.ChannelId, remote.Id)
+	}
+	return remote, nil
+}
+
+// GetSharedChannelRemote fetches a shared channel by shared_channel_remote_id.
+func (s SqlChannelStore) GetSharedChannelRemote(remoteId string) (*model.SharedChannelRemote, error) {
```

Comment:
I think the argument here should be called `id` instead of `remoteId`. SharedChannelRemote does not have a field called `remoteId` so `remoteId` could be confused with `RemoteClusterId`

Reply:
Done.  https://github.com/mattermost/mattermost-server/pull/16065/commits/5b182d5c65ca237d5bdcdb5ef13bf9bf1b293079